test/integration/scheduler/predicates_test.go

```diff
@@ -1055,6 +1055,28 @@ func TestUnschedulablePodBecomesSchedulable(t *testing.T) {
 				return err
 			},
 		},
+		{
+			name: "node gets updated",
+			init: func(cs kubernetes.Interface) error {
+				node, err := createNode(cs, "node-1", nil)
```

Comment:
should this new node name be unique via a uuid?
example usage here https://github.com/kubernetes/kubernetes/blob/323f34858de18b862d43c40b2cced65ad8e24052/test/e2e_node/pods_container_manager_test.go#L26

Reply:
yeah, that would be nice, but I need to access the node name both in the `init` and `update` function. Do you have any suggestion on how to do that?

I guess we could add `nodeName` as a field of the test struct, but that would affect all the other tests.

Reply:
@nodo I think you can make `nodeName` as a `*string` like this

```diff
diff --git a/test/integration/scheduler/predicates_test.go b/test/integration/scheduler/predicates_test.go
index d61ae1a94ca..8f7078750a3 100644
--- a/test/integration/scheduler/predicates_test.go
+++ b/test/integration/scheduler/predicates_test.go
@@ -22,6 +22,7 @@ import (
 	"testing"
 	"time"
 
+	"github.com/google/uuid"
 	v1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
@@ -1039,6 +1040,7 @@ var (
 )
 
 func TestUnschedulablePodBecomesSchedulable(t *testing.T) {
+	var nodeName *string
 	tests := []struct {
 		name   string
 		init   func(kubernetes.Interface) error
@@ -1051,7 +1053,7 @@ func TestUnschedulablePodBecomesSchedulable(t *testing.T) {
 				Name: "pod-1",
 			},
 			update: func(cs kubernetes.Interface) error {
-				_, err := createNode(cs, "node-1", nil)
+				_, err := createNode(cs, *nodeName, nil)
 				return err
 			},
 		},
@@ -1061,6 +1063,9 @@ func TestUnschedulablePodBecomesSchedulable(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			testCtx := initTest(t, "scheduler-informer")
 			defer testutils.CleanupTest(t, testCtx)
+
+			uuid := uuid.New().String()
+			nodeName = &uuid
 			if tt.init != nil {
 				if err := tt.init(testCtx.ClientSet); err != nil {
 					t.Fatal(err)

```

Reply:
Reminder: in k8s, usually after adding a new import, we run `hack/update-bazel.sh` to make bazel happy.

Reply:
Thanks for the suggestion! I will give it a go!

Reply:
I have added a separate commit where I randomize the node and pod names and updated `bazel`.

Reply:
Node names don't need to be unique. Each test case is already isolated and there is a proper cleanup for each.

Reply:
Isn't a clash still possible in case of parallel tests?

Reply:
This tests run serially. The etcd is shared, so it wouldn't be safe to run them in parallel.

Reply:
Right, I thought we wanted to be robust for that, just in case in the future we want to enable parallelism.

Reply:
@neolit123, @Huang-Wei WDYT?

Reply:
Other tests could run in parallel precisely because they can work over the same set of nodes.
But since in this case we have different initial node setups, we can't make them run in parallel ever.
Because of that, we should also try to keep the amount of setup and number of test cases low.

Reply:
I am not 100% convinced. If every node has a unique name, why these tests could not be run in parallel? Each test would create / update / delete a node which won't clash with others. What am I missing?

Reply:
because one of the tests assumes that there are no nodes at the beginning :)

Reply:
aha! now I am convinced 😄 thanks @alculquicondor !

@neolit123 / @Huang-Wei do you agree to come back to hardcoded node names?